Tests/RegexTests/MatchTests.swift

```diff
@@ -915,6 +914,14 @@ extension RegexTests {
       #"\d+(*negative_lookahead: dollars)"#,
       input: "Price: 100 pesos", match: "100")
 
+    // More complex lookaheads
+    firstMatchTest(
+      #"(?=.*e)(?=.*o)(?!.*z)"#,
+      input: "hello", match: "")
+    firstMatchTest(
+      #"^(?=.*e)(?=.*o)(?!.*h)"#,
+      input: "hello", match: nil)
```

Comment:
Similarly, what could be matched by this?

Reply:
This is the negative version of the previous test, since it contains a negative lookahead for `h`, which is present in the input.

Reply:
I'll add some positives though